### ISSUE TITLE
Crash on First Run

### ISSUE THREAD

**Issue by DON-MAC-256** [cloudhead/nakamoto#111]
Running the node for the first time results in a crash. The root of the issue lies in the logic for determining the success of the `event::Subscriber::publish` function.

```rust
impl<T: Clone> Subscriber<T> {
    pub fn publish(&self, event: T) -> bool {
        let mut subs = self.subscribers.lock().unwrap();
        subs.retain(|s| s.try_send(event.clone()).is_ok());
        subs.is_empty().not()
    }
}
```

If there are no subscribers at the time of this call, then failure is returned. This is the case for the `loading` subscriber in the client.

Workaround:

```diff
impl<T: Clone> Subscriber<T> {
    pub fn publish(&self, event: T) -> bool {
        let mut subs = self.subscribers.lock().unwrap();
+++     if subs.is_empty() {
+++         return true;
+++     }
        subs.retain(|s| s.try_send(event.clone()).is_ok());
        subs.is_empty().not()
    }
}
```

**Comment by cloudhead** [comment]
Nice catch, yeah this was recently added and indeed interrupts the node if there is no subscriber. I need to however found a proper solution.

**Comment by cloudhead** [comment]
Potential fix here: https://github.com/cloudhead/nakamoto/pull/114

**Comment by cloudhead** [comment]
Closed with #114
